sgx/trusted/src/ecall/kmyth_enclave_retrieve_key.cpp

```diff
@@ -55,19 +56,48 @@ int kmyth_enclave_retrieve_key_from_server(uint8_t * client_private_bytes,
   }
   kmyth_sgx_log(LOG_DEBUG, "unmarshalled server certificate (to X509)");
 
+  unsigned char *retrieve_key_result = NULL;
+  size_t retrieve_key_result_len = 0;
+  unsigned char *retrieve_key_result_id = NULL;
+  size_t retrieve_key_result_id_len = 0;
+
   ret_val =
     enclave_retrieve_key(client_sign_privkey, server_cert, server_host,
-                         server_host_len, server_port, server_port_len);
+                         server_host_len, server_port, key_id, key_id_len,
+                         &retrieve_key_result_id, &retrieve_key_result_id_len,
+                         &retrieve_key_result, &retrieve_key_result_len);
   if (ret_val)
   {
     kmyth_sgx_log(LOG_ERR,
                   "enclave_retrieve_key() wrapper function call failed");
     return EXIT_FAILURE;
   }
 
+  char msg[MAX_LOG_MSG_LEN] = { 0 };
+
+  snprintf(msg, MAX_LOG_MSG_LEN, "Retrieved into enclave key with ID: %.*s",
+           (int) retrieve_key_result_id_len, retrieve_key_result_id);
```

Comment:
This int cast should be checked because the id_len is a size_t

Reply:
Yes, retrieve_key_result_id_len is of type size_t (unsigned int) and is cast to int because the '.*' precision value in the format string expects an int. I'm not sure what to check, because the uint -> int should never be negative result. If it was, however, the precision would be ignored, which would negate the intended protection against a potentially unterminated string.

The way I tried to fix this was to check that the string parsed out of the KMIP response indeed has the expected length in the enclave_retrieve_key() function and then removed the precision (and hence the need for the cast) from the format string in both that function and here since we have verified with the length check that the string is terminated.

Reply:
I added a comment above the string length checks noting the restriction of this implementation handling the Key ID as a string and not supporting '\0' as a valid character within the Key ID value itself. As the documentation in the header and .edl files defines 'Key ID' parameters as string values, my thought was that they did not all need to include this further clarification.